core/pluginz/ezpz_pluginz/__cli__.py

```diff
@@ -0,0 +1,706 @@
+    ("delete-registry", "Delete the local plugin registry cache"),
+  ]
+
+  for cmd, desc in commands:
+    logger.info(f"  {cmd:<12} {desc}")
+
+  logger.info("")
+  logger.info("EXAMPLES:")
+  logger.info("  ezplugins list                    # Show all available plugins")
+  logger.info("  ezplugins find database           # Search for database-related plugins")
+  logger.info("  ezplugins add my-plugin           # Install and mount 'my-plugin'")
+  logger.info("  ezplugins add my-plugin --no-auto-mount  # Install without mounting")
+  logger.info("  ezplugins help add                # Show detailed help for 'add' command")
+  logger.info("")
+
+  logger.info("CONFIGURATION:")
+  logger.info("  • Configuration file: ezpz.toml (created automatically)")
+  logger.info("  • Registry cache: ~/.ezpz/plugins/registry.json")
+  logger.info("  • Environment variables:")
+  logger.info("    - GITHUB_PAT or GITHUB_TOKEN (for registering/updating plugins)")
+  logger.info("")
+
+  logger.info("For detailed help on any command, use: ezplugins help <command>")
+
+
+@app.command(name="mount")
+def mount() -> None:
+  mount_plugins()
+
+
+@app.command(name="unmount")
+def unmount() -> None:
+  unmount_plugins()
+
+
+@app.command(name="health")
+def health() -> None:
+  remote_reg = PluginRegistryAPI()
+  try:
+    response = remote_reg.check_health()
+    logger.info(response)
+  except Exception as e:
+    logger.info("Health check failed")
+    raise typer.Exit(1) from e
+
+
+@app.command(name="register")
+def register(
+  plugin_path: str = typer.Argument(..., help="Path to the plugin to register"),
+) -> None:
+  config = load_config()
+  if not config:
+    logger.error("Could not load ezpz.toml configuration")
+    raise typer.Exit(1)
+
+  local_registry = LocalPluginRegistry()
+  if not local_registry.fetch_and_update_registry():
+    logger.warning("Failed to refresh local plugin registry, continuing with cached data")
+
+  plugin_info = find_plugin_in_path(plugin_path, config.include_str_paths)
+  if plugin_info is None:
+    logger.error(f"No plugin found at path: {plugin_path}")
+    logger.info("Make sure the path contains a plugin with a register_plugin() function in the module entry i.e '__init__.py'")
+    logger.info(f"Searched in configured include paths: {config.include_str_paths}")
+    raise typer.Exit(1)
+
+  if local_registry.is_plugin_registered(plugin_info.name):
+    logger.info(f"Plugin '{plugin_info.name}' is already registered")
+    logger.info("Skipping registration")
+    return
+
+  github_pat = get_github_pat()
+  api = PluginRegistryAPI()
+  success = api.register_plugin(plugin_info, github_pat)
+
+  if success:
+    logger.info(f"Successfully registered '{plugin_info.name}'")
+    local_registry.fetch_and_update_registry()
+  else:
+    logger.error(f"Failed to register '{plugin_info.name}'")
+    raise typer.Exit(1)
+
+
+@app.command(name="update")
+def update_plugin(
```

Comment:
i think `push` might be more clear than `update` .. seems this is for pushing a plugin to the registry?